pkg/cli/codearea.go

```diff
@@ -196,6 +200,71 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a simple abbreviation that ignores word boundaries.
+func (w *codeArea) expandSimpleAbbr() {
+	var abbr, full string
+	// Find the longest matching abbreviation.
+	w.Abbreviations(func(a, f string) {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr(triggerRune rune, triggerLen int) {
+	// We only do small word expansions at the end of the line and only if
+	// there are characters in the insert buffer other than the trigger rune.
+	c := &w.State.Buffer
+	if len(w.inserts) <= triggerLen || c.Dot < len(c.Content) {
+		return
+	}
+
+	inserts := w.inserts[:len(w.inserts)-triggerLen] // ignore abbr trigger rune
+	var abbr, full string
+	// Find the longest matching abbreviation.
+	w.SmallWordAbbreviations(func(a, f string) {
+		if len(a) >= len(w.inserts) || len(a) < len(abbr) {
+			return // check the next abbreviation
+		}
+
+		// Verify the abbreviation is in the insert buffer.
+		if !strings.HasSuffix(inserts, a) {
+			return
+		}
+		// Verify the trigger rune creates a word boundary.
+		r2, _ := utf8.DecodeLastRuneInString(a)
```

Comment:
There is no `r`, so `r2` can be just called `r`.

Reply:
Yes. This is called `r2` because an earlier revision did have an `r1` var.